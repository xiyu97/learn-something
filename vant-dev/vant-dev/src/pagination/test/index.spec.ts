import { mount } from '../../../test';
import { Pagination } from '..';

test('should render prev-text、next-text slot correctly', () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 50,
      showPageSize: 5,
    },
    slots: {
      'prev-text': () => 'Custom PrevText',
      'next-text': () => 'Custom NextText',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render page slot correctly', () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 50,
      showPageSize: 5,
    },
    slots: {
      page: ({ text }) => `foo ${text}`,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
