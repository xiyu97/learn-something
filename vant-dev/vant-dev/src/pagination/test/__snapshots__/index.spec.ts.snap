// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page slot correctly 1`] = `
<ul class="van-pagination">
  <li class="van-pagination__item van-pagination__prev van-hairline">
    Previous
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    foo 1
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    foo 2
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    foo 3
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    foo 4
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    foo 5
  </li>
  <li class="van-pagination__item van-pagination__next van-hairline">
    Next
  </li>
</ul>
`;

exports[`should render prev-text„ÄÅnext-text slot correctly 1`] = `
<ul class="van-pagination">
  <li class="van-pagination__item van-pagination__prev van-hairline">
    Custom PrevText
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    1
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    2
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    3
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    4
  </li>
  <li class="van-pagination__item van-pagination__page van-hairline">
    5
  </li>
  <li class="van-pagination__item van-pagination__next van-hairline">
    Custom NextText
  </li>
</ul>
`;
